pypika/terms.py

```diff
@@ -776,15 +776,25 @@ def replace_table(  # type:ignore[return]
 
 
 class Array(Tuple):
-    def get_sql(self, **kwargs: Any) -> str:
-        dialect = kwargs.get("dialect", None)
-        values = ",".join(term.get_sql(**kwargs) for term in self.values)  # type:ignore[union-attr]
+    def __init__(self, *values: Any) -> None:
+        super().__init__(*values)
+        self.original_value = list(values)
 
-        sql = "[{}]".format(values)
-        if dialect in (Dialects.POSTGRESQL, Dialects.REDSHIFT):
-            sql = "ARRAY[{}]".format(values) if len(values) > 0 else "'{}'"
+    def get_sql(self, parameterizer: Parameterizer | None = None, **kwargs: Any) -> str:
+        if parameterizer is None or not parameterizer.should_parameterize(self.original_value):
+            dialect = kwargs.get("dialect", None)
+            values = ",".join(
+                term.get_sql(**kwargs) for term in self.values
+            )  # type:ignore[union-attr]
 
-        return format_alias_sql(sql, self.alias, **kwargs)
+            sql = "[{}]".format(values)
+            if dialect in (Dialects.POSTGRESQL, Dialects.REDSHIFT):
+                sql = "ARRAY[{}]".format(values) if len(values) > 0 else "'{}'"
```

Comment:
Why format 2 times for postgres and redshift? 
Looks like it should if/else, not if with overriding

Reply:
Nice catch! This is the old code, I just moved it under `if`. I'll fix it later since I don't want to reset your approve with a commit.